nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -126,29 +134,38 @@ LifecycleManager::destroyLifecycleServiceClients()
   }
 }
 
-void
+bool
+LifecycleManager::changeStateForNode(const std::string & node_name, std::uint8_t transition)
+{
+  if (!node_map_[node_name]->change_state(transition) ||
```

Comment:
I'd split these two checks into separate if statements. While this works they are logically separate. The `change_state` call does work and the `get_state` call checks if it was successful.

Reply:
That's how I originally had it, just thought I could reduce some lines of code. I'm okay with separating though

Reply:
Why have the second check?